periodic/periodic.go

```diff
@@ -238,13 +267,19 @@ func runOne(id int, funcTimes *stats.Histogram, sleepTimes *stats.Histogram, num
 		i++
 		// if using QPS / pre calc expected call # mode:
 		if useQPS {
-			if i >= numCalls {
+			if numCalls > 0 && i >= numCalls {
 				break // expected exit for that mode
 			}
 			elapsed := time.Since(start)
-			// This next line is tricky - such as for 2s duration and 1qps there is 1
-			// sleep of 2s between the 2 calls and for 3qps in 1sec 2 sleep of 1/2s etc
-			targetElapsedInSec := (float64(i) + float64(i)/float64(numCalls-1)) / perThreadQPS
+			var targetElapsedInSec float64
+			if numCalls > 0 {
+				// This next line is tricky - such as for 2s duration and 1qps there is 1
+				// sleep of 2s between the 2 calls and for 3qps in 1sec 2 sleep of 1/2s etc
+				targetElapsedInSec = (float64(i) + float64(i)/float64(numCalls-1)) / perThreadQPS
+			} else {
+				// Calculate the target elapsed when in endless execution
+				targetElapsedInSec = float64(i) / perThreadQPS
+			}
 			targetElapsedDuration := time.Duration(int64(targetElapsedInSec * 1e9))
 			sleepDuration := targetElapsedDuration - elapsed
 			log.Debugf("%s target next dur %v - sleep %v", tIDStr, targetElapsedDuration, sleepDuration)
```

Comment:
let's put that for the case we'd be sleeping (also let's ^C stop right away instead of after sleep) : 
```golang
			select {
			case <-c:
				break MainLoop
			case <-time.After(sleepDuration):
				// continue normal execution
			}
```